src/rust_connection/mod.rs

```diff
@@ -718,37 +694,23 @@ fn write_setup(
             Err(e) => return Err(e),
         }
     }
-    // Flush the buffer (note that `flush` is non-blocking
-    // and can return `WouldBlock` even after polling).
```

Comment:
Why is the flush no longer necessary?

Reply:
Because buffering is not used there.

Reply:
Hm. So this now forbids any kind of buffering in the writer? Perhaps this should still flush... dunno... I'll let you decide